packages/cdk8s/lib/yaml.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs';
+import * as YAML from 'yaml';
+import { execSync } from 'child_process';
+import { parse as urlparse } from 'url';
+import * as os from 'os';
+import * as path from 'path';
+
+/**
+ * YAML utilities.
+ */
+export class Yaml {
+  /**
+   * Saves a set of objects as a multi-document YAML file.
+   * @param filePath The output path
+   * @param docs The set of objects
+   */
+  public static save(filePath: string, docs: any[]) {
+    // convert each resource to yaml and separate with a '---' line
+    const data = docs.map(r => YAML.stringify(r)).join('---\n');
+    fs.writeFileSync(filePath, data, { encoding: 'utf-8' });
+  }
+
+  /**
+   * Saves a set of YAML documents into a temp file (in /tmp)
+   *
+   * @returns the path to the temporary file
+   * @param docs the set of documents to save
+   */
+  public static tmp(docs: any[]): string {
+    const tmpdir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk8s-'));
+    const filePath = path.join(tmpdir, 'temp.yaml');
+    this.save(filePath, docs);
```

Comment:
`this` inside a static method? does this translate well to Java/C#?

```suggestion
    Yaml.save(filePath, docs);
```

Reply:
Oh wait, this doesn't actually get **executed** in other languages. Still though...

Reply:
Yes, totally works in JavaScript/TypeScript (this is the implementation, no it's not relevant for java/C#).